splink/comparison_helpers.py

```diff
@@ -0,0 +1,433 @@
+import re
+
+import altair as alt
+import jellyfish
+import pandas as pd
+import phonetics
+
+
+def jaccard_similarity(str1, str2):
+    set1 = set(str1)
+    set2 = set(str2)
+    return len(set1 & set2) / len(set1 | set2)
+
+
+def comparator_score(str1, str2, decimal_places=2):
+    """Helper function to give the similarity between two strings for
+    the string comparators in splink.
+
+    Examples:
+        ```py
+        import splink.comparison_helpers as ch
+
+        ch.comparator_score("Richard", "iRchard")
+        ```
+    """
+
+    scores = {}
+
+    # Levenshtein distance
+    lev_dist = jellyfish.levenshtein_distance(str1, str2)
+    scores["levenshtein_distance"] = round(lev_dist, decimal_places)
+
+    # Damerau-Levenshtein distance
+    dlev_dist = jellyfish.damerau_levenshtein_distance(str1, str2)
+    scores["damerau_levenshtein_distance"] = round(dlev_dist, decimal_places)
+
+    # Jaro distance
+    jaro_sim = jellyfish.jaro_distance(str1, str2)
+    scores["jaro_similarity"] = round(jaro_sim, decimal_places)
+
+    # Jaro-Winkler distance
+    jw_sim = jellyfish.jaro_winkler(str1, str2)
+    scores["jaro_winkler_similarity"] = round(jw_sim, decimal_places)
+
+    # Jaccard similarity
+    jaccard_sim = jaccard_similarity(str1, str2)
+    scores["jaccard_similarity"] = round(jaccard_sim, decimal_places)
```

Comment:
Can we do this with duckdb snippets instead?

Reply:
for example:
```
import duckdb
con = duckdb.connect()

def lev_duckdb(str1, str2):
    sql = f"""
        select levenshtein('{str1}', '{str2}') as lev
    """
    return con.execute(sql).fetch_df()
    
lev_duckdb("rich", "richya")
```

This removes the need to have another import and you can also remove the jaccard function.

Reply:
Actually, you could just write one single SQL string for this

Reply:
Feel free to clean it up, but this works:

```
sql = f"""
    select 
    levenshtein('{str1}', '{str2}') as levenshtein_distance,
    damerau_levenshtein('{str1}', '{str2}') as damerau_levenshtein_distance,
    jaro_similarity('{str1}', '{str2}') as jaro_similarity,
    jaro_winkler_similarity('{str1}', '{str2}') as jaro_winkler_similarity,
    jaccard('{str1}', '{str2}') as jaccard_similarity
"""
```

then just adjust the decimal places.

Reply:
Great idea - done